space-cli/cmd/modules/services/generate.go

```diff
@@ -179,3 +179,75 @@ func GenerateServiceRoute(projectID string) (*model.SpecObject, error) {
 
 	return v, nil
 }
+
+// GenerateServiceRole creates a service role struct
+func GenerateServiceRole(projectID string) (*model.SpecObject, error) {
+	if projectID == "" {
+		if err := input.Survey.AskOne(&survey.Input{Message: "Enter Project ID:"}, &projectID); err != nil {
+			return nil, err
+		}
+	}
+
+	roleID := ""
+	if err := input.Survey.AskOne(&survey.Input{Message: "Enter Role ID:"}, &roleID); err != nil {
+		return nil, err
+	}
+
+	serviceID := ""
+	if err := input.Survey.AskOne(&survey.Input{Message: "Enter serviceID:"}, &serviceID); err != nil {
+		return nil, err
+	}
+
+	Type := ""
+	if err := input.Survey.AskOne(&survey.Input{Message: "Enter Type:"}, &Type); err != nil {
+		return nil, err
+	}
+
+	rules := make([]model.Rule, 0)
+	repeat := "YES"
+	for repeat == "YES" {
+		apiGroup := ""
+		if err := input.Survey.AskOne(&survey.Input{Message: "Enter APIGroups:"}, &apiGroup); err != nil {
+			return nil, err
+		}
+		apiGroups := strings.Split(apiGroup, ",")
```

Comment:
Handle the case, when the last element after comma is not specified
a,b,c,
you array will [a,b,c,""]